### ISSUE TITLE
`@inject` breaks `inspect.iscoroutinefunction`

### ISSUE THREAD

**Issue by burritoatspoton** [ets-labs/python-dependency-injector#574]
After upgrade to 4.39.0, while `asyncio.coroutines.iscoroutinefunction` is preserved, `inspect.iscoroutinefunction` is not -- tested on Python 3.10.4:
```python
from dependency_injector.wiring import inject
import inspect

async def foo(): pass

@inject
async def bar(): pass

print(inspect.iscoroutinefunction(foo))
print(inspect.iscoroutinefunction(bar))
```
This e.g. makes FastAPI incorrectly recognize `bar` as a "normal" function when used with `dep = fastapi.Depends(bar)` syntax in router dependencies.

**Comment by rmk135** [comment]
Hi @burritoatspoton ,

Thanks a lot for reporting the issue. I have Cythonized some parts of ``@inject`` decorator and it caused the problem. Seems like the problem is known and the roots go back to 2018: https://github.com/cython/cython/issues/2273.

Working on the fix.

Best,
Roman

**Comment by rmk135** [comment]
Fixed in ``4.39.1``. The fix should be safe and most of the applied optimizations from ``4.39.0`` are still in place.

PS: here's the commit with the hotfix summary highlighted: https://github.com/ets-labs/python-dependency-injector/commit/cfed30cf076baee02616c628f31a5b0341d054ac#diff-4af6e647307dbe71b243ef06bbbefaafa5c03b9a57b92bc8106cdd4a3e96a6e7R901-R913

**Comment by burritoatspoton** [comment]
Thank you for the quick fix!
Good day
Rafal
